packages/perseus/src/widgets/python-program.tsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * This widget is for embedding Khan Academy Python programs.
+ */
+
+import {StyleSheet, css} from "aphrodite";
+import PropTypes from "prop-types";
+import * as React from "react";
+import _ from "underscore";
+
+import {getDependencies} from "../dependencies";
+import {isFileProtocol} from "../util/mobile-native-utils";
+import {toAbsoluteUrl} from "../util/url-utils";
+
+import type {WidgetExports} from "../types";
+
+function getUrlFromProgramID(programID: any) {
+    const {InitialRequestUrl} = getDependencies();
+
+    const path = `/python-program/${programID}/embedded`;
+
+    return toAbsoluteUrl(path);
+}
+
+/* This renders the program in an iframe. */
+class PythonProgram extends React.Component<any> {
```

Comment:
This seems to be a really simple component. Is it possible to write it as a functional component?  We are working to write new code in modern React and want to avoid writing "legacy" code when we can.

Reply:
I tried it, but the `renderQuestion()` function used in tests is not compatible with functional components. It appears all Widgets are currently class components. Given that, do you think it would be okay to leave it as a class component?

Reply:
Ah! Yes.  We'll break that new ground when we have time.  I didn't realize we haven't migrated any actual widgets to functional React components.